tutorials/integration.md

```diff
@@ -824,68 +856,89 @@ contract("CaptureTheFlag", async accounts => {
+&nbsp;
 
-[source.javascript]
-```
-		const gsnInstance = await gsnTestEnv.startGsn(blockchain);
+The `GsnTestEnvironment.startGsn` command starts GSN on the provided blockchain. 
+
+```javascript
+                let env = await GsnTestEnvironment.startGsn('localhost')
 ```
 
+&nbsp;
+
 The various addresses GSN uses on the blockchain are stored in 
-`gsnInstance.deploymentResult`. Use them to initialize our contracts.
+`env.deploymentResult`. Use them to initialize our contracts.
 
-[source.javascript]
-```
-		const flag = await 
-			CaptureTheFlag.new(gsnInstance.deploymentResult.forwarderAddress)
+```javascript
 
-		const paymaster = await NaivePaymaster.new()
-		await paymaster.setRelayHub(
-			gsnInstance.deploymentResult.relayHubAddress)
-		await paymaster.send(1e17)
-		await paymaster.setTarget(flag.address)
-```
+		const { forwarderAddress } = env.deploymentResult
+                .
+                .
+                .
+                const factory = new ethers.ContractFactory(
+                        CaptureTheFlag.abi,
+                        CaptureTheFlag.bytecode,
+                        deploymentProvider.getSigner())
 
-This part is the same as what we did for manual testing, except for the 
-GSN parameters coming from `gsnInstance` instead of being entered manually.
 
-[source.javascript]
+                const flag = await factory.deploy(forwarderAddress)
+                await flag.deployed()
+                .
+                . 
+                .
+                const paymasterFactory = new ethers.ContractFactory(
+                        NaivePaymaster.abi,
+                        NaivePaymaster.bytecode,
+                        deploymentProvider.getSigner()
+                )
+
+                const paymaster = await paymasterFactory.deploy()
+                await paymaster.deployed()
+                await paymaster.setTarget(flag.address)
+                await paymaster.setRelayHub(env.deploymentResult.relayHubAddress)
+                await paymaster.setTrustedForwarder(forwarderAddress)
+
+                web3.eth.sendTransaction({
+                        from:accounts[0],
+                        to:paymaster.address,
+                        value:1e18})
 ```
-		const gsnConfigParams = {
-			gasPriceFactorPercent: 70,
-			methodSuffix: '_v4',
-			jsonStringifyRequest: true,
-			chainId: '*',
-			relayLookupWindowBlocks: 1e5,
-			preferredRelays: [ gsnInstance.relayUrl ],
-			relayHubAddress: 
-				gsnInstance.deploymentResult.relayHubAddress,
-			stakeManagerAddress: 
-				gsnInstance.deploymentResult.stakeManagerAddress,
-			paymasterAddress: paymaster.address
-		}
+
+&nbsp;
+
+The GSN provider is created the same way we did for manual testing, except for the 
+GSN parameters coming from `env` instead of being entered manually.
+
+```javascript
+                let gsnProvider = await
+                    new RelayProvider(web3provider, {
```

Comment:
new 2.1 syntax:

`= await RelayProvider.newProvider( { provider: web3.provider, config: { } ).init()

Reply:
1. When is 2.1 going to be released to npm?
2. Do we need to change the version in the contracts too, or does 2.1 apply only to the JavaScript?

Reply:
1. It was just released a few hours ago
2. No, the contracts remain exactly the same for now

Reply:
I get this error:
```
     TypeError: Cannot read property 'host' of undefined
````

I use this:
```
                       await RelayProvider.newProvider({
                                provider: web3.provider,
                                config: {
                                        forwarderAddress,
                                        paymasterAddress: paymaster.address
                                }
                        } ).init()
```

Can you tell me what I'm missing?

Reply:
web3.currentProvider